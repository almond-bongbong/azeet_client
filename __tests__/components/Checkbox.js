import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Checkbox from 'components/Checkbox/Checkbox';

describe('Checkbox', () => {
  it('matches snapshot', () => {
    const utils = render(<Checkbox onChange={() => null} id="checkbox0" />);

    expect(utils.container).toMatchSnapshot();
  });

  it('click and check', () => {
    const onChange = jest.fn();
    const { getByDisplayValue } = render(<Checkbox onChange={onChange} value="value" id="checkbox1" label="체크박스" />);
    const checkbox = getByDisplayValue('value');

    fireEvent.click(checkbox);

    expect(onChange).toBeCalled();
  });
});
