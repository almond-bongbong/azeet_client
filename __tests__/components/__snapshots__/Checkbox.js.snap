// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox matches snapshot 1`] = `
<div>
  <div
    class="Checkbox__BasicCheckboxStyle-sc-1p1hzqe-0 idCwik"
    label=""
  >
    <label
      for="checkbox0"
    >
      <input
        id="checkbox0"
        type="checkbox"
        value=""
      />
      <span />
    </label>
  </div>
</div>
`;
